subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -518,6 +529,7 @@ private void collectSuperConfigs(Configuration configuration, Set<Configuration>
 
     @Override
     public Configuration defaultDependencies(final Action<? super DependencySet> action) {
+        warnOnDeprecatedUsage("defaultDependencies(Action)", ProperMethodUsage.DECLARABLE_AGAINST);
```

Comment:
Looking at this now, I question whether this method is useful for resolvable configurations as well? Take for example
```
configurations {
   resolvable("foo")
   bucket("bar")
   bucket("baz")
   foo.extendsFrom(bar, baz)
}
```
How would a user express that they want foo to have some dependencies by default if bar and baz do not have any dependencies?

Reply:
I don't think this is necessary.  In this case, they could do:

```
configurations {
   resolvable("foo")
   bucket("bar")
   bucket("baz")
   bucket("myDefaults")
   myDefaults.extendsFrom(bar, bar)
   foo.extendsFrom(myDefaults)
}
```

and add the defaults to `myDefaults` instead.

This would be easier to do if we added some richer convenience methods to the API, but I think the scenario is already complex enough that most users won't bump into it.  It's better for this method to remain declarable-only, with the other methods that add dependencies.